pasteit/routes/index.js

```diff
@@ -1,24 +1,45 @@
 var express = require('express');
 var bodyParser = require("body-parser");
 var router = express();
+var redis = require("redis");
+var client = redis.createClient();
+var flatten = require("flat")
 
-router.use(bodyParser.urlencoded({ extended : false }));
-router.use(bodyParser.json());
-router.use(function(err, req, res, next) {
-    console.error(err.stack);
-    res.status(500).send("Something broke!");
-});
+var getNewHash = function() {
+    return (Math.floor(Math.random() * 1e15) + new Date().getMilliseconds()).toString(36);
+};
 
 router.get('/', function(req, res, next) {
     res.render('index', { title: 'Express' });
 });
 
 router.post('/submitPaste', function(req, res, next) { 
-    var l = req.body.language;
-    var p = req.body.paste;
-    var m = req.body.mime;
-    
-    res.render('pasted', { language: l, pastedcontent: p, mime: m });
+    var new_paste = {
+        "language": req.body.language,
+        "paste": req.body.paste,
+        "mime" : req.body.mime
+    }
+
+    var hash = getNewHash();
+
+    client.hmset(hash, new_paste);
+
+    res.redirect("/"+hash);  
+});
+
+router.get("/:hash/", function(req, res) {
+    var hash = req.params.hash;
+
+    var result = client.hgetall(hash, function(err, reply) {
```

Comment:
What will hgetall return if hash is not found?

Reply:
oh right, we would need a 404 or sth like that. I'll look into that.

Reply:
Upon entering a wrong hash an error page is shown.

![grafik](https://user-images.githubusercontent.com/8678523/31322556-77eb9338-ac9a-11e7-9049-5b56303461ec.png)